### DISCUSSION TITLE
boxbilling to Spanish?

### DISCUSSION THREAD

**Discussion by fjgroup** [boxbilling/boxbilling#1224] [General]
Hello,

Is it possible to put boxbilling in Spanish.
I downloaded the PO file and using "POEDIT" generated the MO file.
I have carried out the entire procedure, the system tells me that there are two languages, but it always stays in the English language.

Realice todos estos pasos, pero no logre lo deseado. Quien podria ayudarme?

BoxBilling in your language
Although BoxBilling displays in U.S. English by default, the software has the built-in capability to be used in any language. Follow instructions bellow to install new language.

Check if your language translation file is available at [BoxBilling translations repository](https://www.transifex.com/projects/p/boxbilling/)
Download language files and place them to /home/xxxxx/public_html/xxxx/bb-locale
Language will be automatically available and can be changed in client area
Read [Installing BoxBilling in Your Language](http://docs.boxbilling.com/en/latest/reference/language.html) to learn how to install files that will transform BoxBilling so it displays in your language.
Translations can be made with http://www.poedit.net/ software
Changing default language
Default language can be set in bb-config.php parameter define('BB_LOCALE', 'en_US');
Thank you.

Muchas Gracias.-

**Comment by treborin** [comment]
Hola, @fjgroup. Qué versión de Boxbilling estás usando para traducir?

**Reply by fjgroup** [reply]
Estoy usando esta Powered by [BoxBilling 4.22.1.3]

gracias.-

**Reply by fjgroup** [reply]
@treborin  Posee Usted el BoxBilling en español? Que versión esta usando? Gracias.-

**Reply by treborin** [reply]
No, @fjgroup. En versiones antiguas (inseguras) sí era posible la traducción, pero en esa versión en concreto, por ahora, no me funciona. Estoy viendo cómo resolverlo, aunque dispongo de poco tiempo para ello. Si doy con la solución, lo comunico aquí inmediatamente. Estas últimas versiones no dejan de ser versiones beta, con mucho código nuevo aportado, a falta de unificarlo todo.

**Comment by alchemist-153** [comment]
Hola @fjgroup @treborin , por fin pude hacer que BoxBilling se muestre en español y otros idiomas. La traducción al español de España "Spanish (Spain, es_ES)" que bajé de Transifex solo tiene 34% de avance, en cambio las traducciones que aparecen como "Spanish (es)" y "Spanish (Latin America, es_419)" están al 100%, lamentablemente en estos dos últimos casos BoxBilling no las reconoce, probablemente porque el código del lenguaje no sigue el formato xx_XX, pero dependiendo de lo que ustedes requieran una opción es renombrar el directorio "es" o "es_419" a otro, por ejemplo "es_ES" o "es_MX", y entonces funciona bien. Adjunto los lenguajes mencionados:
[es.zip](https://github.com/boxbilling/boxbilling/files/8422370/es.zip)
[es_419.zip](https://github.com/boxbilling/boxbilling/files/8422371/es_419.zip)

Aunque el lenguaje esté 100% traducido hay algunas partes que se muestran en inglés, supongo que la interfaz de usuario ha sufrido cambios que no se han reflejado en los archivos de lenguaje, los cuales tienen varios años que no se actualizan. La explicación detallada de todo la puse en este hilo: https://github.com/boxbilling/boxbilling/issues/1016

**Reply by treborin** [reply]
Gracias, @alchemist-153, renombrando es a es_Es, después de borrar caché del navegador, lo reconoce perfectamente (uso la v. 4.22.1.3). Ahora es cuestión de ir localizando lo que no se ha incluido en la traducción e ir actualizándolo todo.
Un saludo
